cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -40,6 +41,29 @@ void getInputClassCardinality(
   detail::getInputClassCardinality(groundTruth, nSamples, stream, minLabel, maxLabel);
 }
 
+/**
+ * @brief use this to allocate output matrix size
+ * size of matrix = (maxLabel - minLabel + 1)^2 * sizeof(int)
+ * @param handle: the raft handle.
+ * @param groundTruth: device 1-d array for ground truth (num of rows)
+ * @param nSamples: number of elements in input array
+ * @param minLabel: [out] calculated min value in input array
+ * @param maxLabel: [out] calculated max value in input array
+ */
+template <typename T, typename IdxType, typename LayoutPolicy, typename AccessorPolicy>
+void getInputClassCardinality(
+  const raft::handle_t& handle,
+  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> groundTruth,
+  const raft::host_scalar_view<T>& minLabel,
```

Comment:
Given that `host_scalar_view` is an mdspan, would you consider passing it by value instead of const reference, since mdspan is a view and views are idiomatically passed by value?

(This suggestion generally applies to all the functions in this PR.)